### ISSUE TITLE
Implement Prow

### ISSUE THREAD

**Issue by leodido** [falcosecurity/falco#621]
[Prow](https://github.com/kubernetes/test-infra/tree/master/prow) provides CI features as well as a set of tools to improve developers productivity and to automate workflow as much as possible.

## What it is

The Kubernetes Testing SIG describes Prow as "a CI/CD system built on Kubernetes for Kubernetes that executes jobs for building, testing, publishing and deploying."

Anyway one thing I would like to highlight are its capabilities to inject automation from code commits onward (via [hook](https://github.com/kubernetes/test-infra/blob/master/prow/cmd/README.md#core-components), a scalable staless microservice with plugins that triggers k8s jobs).

So it's not only a CI/CD tool but also a comprehensive GitHub automation tool that can be fine-tuned depending on the needs we have:
- ChatOps
- Fine-grained GH policy/permission/membership mgmt
- PR/merge automation
- Labeling automation
- Mechanism (request cache) to minimize impact on API limits
- Release note management
- Prometheus metrics for monitoring and alerting
- A bunch of other cool things - eg., [cats](https://github.com/kubernetes/test-infra/tree/master/prow/plugins/cat) 😸 

## How it works

Basically via pre-submit, post-submit and periodic jobs.
It can be operated via a kubernetes cluster.

### A bot account

I'd like to have [falconieri](https://github.com/falconieri).
It must be granted owner level access to the GitHub orgs that prow will operate on.

### Deployment

On a subdomain of falco.org - eg., [prow.falco.org](http://prow.falco.org)

## Tools using it

- istio
- k8s
- tekton
- ...

In the upcoming days I'll come with a more details phase 1 plan tailored on our needs.

cc @fntlnz @mfdii

**Comment by leodido** [comment]
Following up the today's office hours and community feedbacks we decided to pursue this path.

I'll come up soon with a plan.

**Comment by fntlnz** [comment]
The issue templates, PR templates and contributing md files are done here: https://github.com/falcosecurity/falco/pull/644

**Comment by fntlnz** [comment]
/assign @markyjackson-taulia 

Assigning Marky since he is doing most of the work for our new CI . 

Marky this was the original issue where we was tracking the CI work.

If the two of you agree we can close this and do an issue for JenkinsX directly.

**Comment by markyjackson-taulia** [comment]
I am fine with closing this

**Comment by leodido** [comment]
Me too.
